pkgs/development/compilers/llvm/3.9/llvm.nix

```diff
@@ -88,6 +97,9 @@ in stdenv.mkDerivation rec {
     ++ stdenv.lib.optionals (isDarwin) [
     "-DLLVM_ENABLE_LIBCXX=ON"
     "-DCAN_TARGET_i386=false"
+  ] ++ stdenv.lib.optionals (buildPlatform != hostPlatform) [
+    "-DCMAKE_CROSSCOMPILING=True"
```

Comment:
Hmm, this is a general cmake thing? This could be generalized to be so added for all derivations with `cmakeFlags` I suppose.